### PR TITLE
Appl 8689 merge updates from base repo

### PR DESCRIPTION
**Pull request by jonobarel-slatescience**
Updated branch with all changes from upstream gree/unity-webview.

### CODE REVIEW

File: build/.gitignore [hunk 1]
```diff
@@ -1,2 +1,6 @@
 Packager/Assets/Plugins
 Packager/Library
+
+#MATIFIC SPECIFIC
```
Comment (lavikabel) [inline build/.gitignore]: why?

File: build/Rakefile [hunk 2]
```diff
@@ -20,7 +20,7 @@
 
 require 'fileutils'
 
-UNITY="/Applications/UnityCurrent/Unity.app/Contents"
+UNITY="/Applications/Unity/Hub/Editor/2019.4.40f1/Unity.app/Contents/MacOS/Unity"
```
Comment (lavikabel) [inline build/Rakefile:23]: Are the changes here yours?

File: plugins/Android/MAD_README.md [hunk 3]
```diff
@@ -12,7 +12,8 @@ This is a custom unity webview fork to allow pausing and resuming unity while th
  	 
 1. `plugins/Android/local.properties` needs to be configured w/ path to Android SDK used by Unity (i.e. `sdk.dir=/Applications/Unity/Hub/Editor/2020.3.30f1/PlaybackEngines/AndroidPlayer/SDK`)
 2. Goto `plugins\Android` and open `install.sh`. (**Don't use `install-nofragment.sh`**)
```
Comment (lavikabel) [inline plugins/Android/MAD_README.md:14]: the nofragment parts are no longer needed, right?

File: plugins/Android/install.sh [hunk 4]
```diff
@@ -1,51 +1,128 @@
...
         ;;
-    '--il2cpp' )
-        SCRIPTING_BACKEND="il2cpp"
+    '--development')
+        MODE="Development"
         ;;
-    '--mono' )
-        SCRIPTING_BACKEND="mono"
+    '--zorderpatch')
+        UNITY='2022.3.32f1'
```
Comment (lavikabel) [inline plugins/Android/install.sh:43]: we dont need the `zorderpatch` flag for us it is always the same unity

File: build/.gitignore (hunk 1 above)
Comment (jonobarel-slatescience) [reply build/.gitignore]: I want the plugin files to be included in the repo.

File: build/Rakefile (hunk 2 above)
Comment (jonobarel-slatescience) [reply build/Rakefile:23]: No. These are from the main repo.

File: plugins/Android/MAD_README.md (hunk 3 above)
Comment (jonobarel-slatescience) [reply plugins/Android/MAD_README.md:14]: The parameter still exists. I should add a note about --development, though.

File: plugins/Android/install.sh (hunk 4 above)
Comment (jonobarel-slatescience) [reply plugins/Android/install.sh:43]: I didn't touch this, only the unity version.

Comment (lavikabel) [reply plugins/Android/install.sh:43]: ok, revert to what gree had, one less conflict in the future.
We are not passing this flag anyway

File: build/.gitignore (hunk 1 above)
Comment (lavikabel) [reply build/.gitignore]: hmmm why?

File: build/.gitignore [hunk 5]
```diff
@@ -1,6 +0,0 @@
-Packager/Assets/Plugins
-Packager/Library
```
Comment (lavikabel) [inline build/.gitignore:2]: why did you remove this file?

File: plugins/Android/MAD_README.md [hunk 6]
```diff
@@ -12,31 +12,46 @@ This is a custom unity webview fork to allow pausing and resuming unity while th
...
-4. Now you can take `WebViewPlugin.aar` and `core-1.6.0.aar.tmpl` that were created in previous section and paste them in `Vendors/unity-webview/dist/package/Assets/Plugins/Android`.
+3. You need to remove old assembly (will appear greyed-out/error) and reimport new assembly (from `Vendors`) under `Assets/AssetsAssembly.asmdf`.
+4. Now you can take `WebViewPlugin-*.aar.tmpl` and `core-1.6.0.aar.tmpl` that were created in previous section and paste them in `Vendors/unity-webview/dist/package/Assets/Plugins/Android`.
 5. You can modify WebViewObject.cs as you like.
 6. If you add more changes to `CUnityPlayerActivity` or `CWebViewPlugin` Build the project, rebuild plugin and redo step 4.
  
 
 ## Deploying new Plugin in git
-1. Copy `WebViewPlugin.aar` and `core-1.6.0.aar.tmpl` from `build/Packager/Assets/Plugins/Android` to `dist/package/Assets/Plugins/Android`
-2. Push to git `WebViewPlugin.aar` and `core-1.6.0.aar.tmpl` from previous section.
```
Comment (lavikabel) [inline plugins/Android/MAD_README.md:40]: Why did you remove those steps?

File: plugins/Android/MAD_README.md [hunk 7]
```diff
@@ -12,31 +12,46 @@ This is a custom unity webview fork to allow pausing and resuming unity while th
...
-3. In your SlateApps's branch - reimport the plugin (delete `packages-lock` entry or remove and reimport from packages manager) - the package's `hash` should change in the packages-lock file
-4. For reimporting the package to unity project delete the relevant section in "Packages/packages-lock.json" under the unity project and reload.
\ No newline at end of file
+1. The `.tmpl` files for `WebViewPlugin` and `core-*` are already in the repo.
+2. After running `install.sh` and recompiling, push the updated file versions to github.
+3. In your SlateApps's branch - edit `Packages/manifest.json` and replace the current hash with that of your latest commit.
+   
+   e.g., on the appropriate package line:
+   > "net.gree.unity-webview": "https://github.com/SlateScience/unity-webview.git?path=/dist/package#ab84fb0517eb7b3a113465a23ed28fdb5ce92446",
```
Comment (lavikabel) [inline plugins/Android/MAD_README.md:54]: please emphasis that  "Packages/packages-lock.json" should change and be pushed

File: plugins/Android/install.sh [hunk 8]
```diff
@@ -1,51 +1,128 @@
...
+then
+    echo 'Looking for Java Home'
+    echo 'From Unity Hub, please install 2022.3.32f1 with the android module.'
+    exit 1
+fi
 
 # options
+TARGET="webview"
 MODE="Release"
 SCRIPTING_BACKEND="mono"
```
Comment (lavikabel) [inline plugins/Android/install.sh]: `SCRIPTING_BACKEND` not in use anymore you can remove

File: plugins/Android/install.sh [hunk 9]
```diff
@@ -1,51 +1,128 @@
...
+
+# MATIFIC SPECIFIC -- commented out this block because it is deleting our CUnityPlayerActivity class.
+## remove CUnityPlayer*.java if UNITY != 5.6.1f1.
+#case $UNITY in
+#'5.6.1f1')
+#    ;;
+#*)
+#    rm -f ${TARGET}/src/main/java/net/gree/unitywebview/CUnityPlayer*.java
+#    ;;
+#esac
```
Comment (lavikabel) [inline plugins/Android/install.sh:106]: please add `END MATIFIC SPECIFIC`

**Review by lavikabel — CHANGES_REQUESTED** [review]
Great work!!1
Not an easy task, you passed every obstacle thrown at you.

File: build/.gitignore (hunk 5 above)
Comment (jonobarel-slatescience) [reply build/.gitignore:2]: Restored the file. Thought it was blocking the tracking of the .tmpl files but something else was doing it.

File: plugins/Android/MAD_README.md (hunk 6 above)
Comment (jonobarel-slatescience) [reply plugins/Android/MAD_README.md:40]: See updated version, the instructions are updated to include these again as I thought the project structure had changed.

File: build/.gitignore (hunk 1 above)
Comment (jonobarel-slatescience) [reply build/.gitignore]: no longer relevant.

**Review by lavikabel — APPROVED** [review]
